components/lib/n3-utils.js

```diff
@@ -0,0 +1,312 @@
+                        ttl2Jsonld.rdfToJsonLd(n3Store, frame, filterBnodeAttrs).then(
+                            function(json) { 
+
+                                if (_.isEmpty(filename)) { 
+                                    return resolve(json);
+                                }
+
+                                fs.writeFile(filename, JSON.stringify(json, null, 2), 
+                                    function(error) { 
+                                        if (!_.isNull(error)) { 
+                                            return reject(errUtils.errorMessage("n3ToTtlToJsonld API was unable to write file " + filename + "!\n", error));
+                                        } 
+
+                                        return resolve(json);
+                                });
+                            });
+                         }, 
+
+                         function(fail) { 
+                             return reject(errUtils.errorMessage('Failed to load ttl!', fail));
+
+                     }).catch(function(e) { 
+                         return reject(errUtils.errorMessage('Exception loading ttl!', e));
+                     });
+            }, 
+
+            function(fail) { 
+                reject(errUtils.errorMessage("n3ToTtlToJsonld error converting N3 store to TTL!", fail));
+    
+        }).catch(function(e) { 
+            return Promise.reject(errUtils.errorMessage('Exception in n3ToTtlToJsonld!', e));
+        });
+    });
+}
+
+/**
+ * Converts an N3 store into JSON-LD. It is an asychronous function.
+ * 
+ * @param store    (required) the N3 store to convert to JSON-LD format
+ * @param frame    (required) a JSON-LD frame
+ * @param filename (optional) an optional file name to write the Turtle result into.
+ *
+ * @return a promise to return either the JSON-LD format or a reject on error
+ */ 
+function n3ToJsonld(n3Store, frame, filename) {
+    if (_.isEmpty(n3Store)) { 
+        return Promise.reject(errUtils.errorMessage("n3ToJsonld API requires an N3 store!"));
+    }
+
+    if (_.isEmpty(frame)) { 
+        return Promise.reject(errUtils.errorMessage("n3ToJsonld API requires a frame!"));
+    }
+
+    var triples = n3Store.find();
+    if (_.isEmpty(triples)) { 
+        return Promise.resolve('');
+    }
+       
+    return new Promise(function(resolve, reject) { 
+        resolve(
+             jsonld.fromRDF( 
+                triples.reduce(
+                    function(dataset, triple) {
+                        var graph = triple.graph || '@default';
+                        dataset[graph] = dataset[graph] || [];
+                        dataset[graph].push({
+                            subject: fromN3(triple.subject),
+                            predicate: fromN3(triple.predicate),
+                            object: fromN3(triple.object)
+                        });
+                        return dataset;
+                    }, {'@default': []}), {})   
```

Comment:
took me a moment to realize that {} was a second argument to a function above...